collectors/proc.plugin/proc_net_dev.c

```diff
@@ -503,6 +504,18 @@ int do_proc_net_dev(int update_every, usec_t dt) {
             else
                 d->virtual = 0;
 
+            // set nic speed if present
+            procfile *df = NULL;
+            snprintfz(buffer, FILENAME_MAX, "/sys/class/net/%s/speed", d->name);
+            if(likely(!d->virtual)) df = procfile_open(buffer, " \t,:|", PROCFILE_FLAG_DEFAULT);
+            if(likely(df)) {
+              df = procfile_readall(df);
+              if(likely(df)) {
+                d->speed_max = str2kernel_uint_t(procfile_word(df, 0));
+              }
+              procfile_close(df);
+            }
+
```

Comment:
3 issues with this code block:

1. We should respect the host prefix of netdata, so that when netdata runs in a container it will be able to find the `/proc` and `/sys` files of the host. This is done like this: https://github.com/netdata/netdata/blob/15408ace0c2a81986026dd66446bb216c9b96eb6/collectors/proc.plugin/proc_net_dev.c#L442-L443

2. All files should pass through `nedata.conf` so that users will be able to override the whole path if they need to. The example above does that.

3. Using `procfile` for that little thing is overkill. You can use this: https://github.com/netdata/netdata/blob/15408ace0c2a81986026dd66446bb216c9b96eb6/collectors/cgroups.plugin/cgroup-network.c#L63-L64

Reply:
updated, much cleaner now